docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+
+This command will only be available within the guild that you specified.
+
+> warn
+> Guild commands update **instantly**. We recommend you use guild commands for quick testing, and global commands when they're ready for public use.
+
+## Updating and Deleting a Command
+
+Slash Commands can be deleted and updated by making `DELETE` and `PATCH` calls to the command endpoint. Those endpoints are
+
+- `applications/<my_application_id>/commands/<command_id>` for global commands, or
+- `applications/<my_application_id>/guilds/<guild_id>/commands/<command_id>` for guild commands
+
+Because Slash Commands have unique names within a scope, we treat `POST` requests for new commands as upserts. That means **making a new command with an already-used name for your application will update the existing command**.
+
+Full documentation of endpoints can be found in [Endpoints](#DOCS_INTERACTIONS_SLASH_COMMANDS/endpoints).
+
+## Receiving an Interaction
+
+When a user uses a Slash Command, your app will receive an **Interaction**. Your app can receive an interaction in one of two ways:
+
+- Via gateway event, `INTERACTION_CREATE` <docs>
+- Via outgoing webhook
+
+These two methods are **mutually exclusive**; you can _only_ receive Interactions one of the two ways. The `INTERACTION_CREATE` gateway event will be handled by the library you are using, so we'll just cover outgoing webhooks.
+
+In your application in the Developer Portal, there is a field on the main page called "Interactions Endpoint URL". If you want to receive Interactions via outgoing webhook, you can set your URL in this field. In order for the URL to be valid, you must be prepared for two things ahead of time:
+
+> info
+> These steps are only necessary for webhook-based Interactions. It is not required for receiving them over the gateway.
+
+1. Your endpoint must be prepared to ACK a `PING` message
+2. Your endpoint must be set up to properly handle signature headers--more on that in [Security and Authorization](#DOCS_INTERACTIONS_SLASH_COMMANDS/security-and-authorization)
+
+If either of these are not complete, we will not validate your URL and it will fail to save.
+
+When you attempt to save a URL, we will send a `POST` request to that URL with a `PING` payload. The `PING` payload has a `type: 1`. So, to properly ACK the payload, return a `200` reponse with a payload of `type: 1`:
+
+```py
+@app.route('/', methods=['POST'])
+def my_command():
+    if request.json["type"] == 1:        
+        return jsonify({
+            "type": 1
+        })
+```
+
+You'll also need to properly set up [Security and Authorization](#DOCS_INTERACTIONS_SLASH_COMMANDS/security-and-authorization) on your endpoint for the URL to be accepted. Once both of those are complete and your URL has been saved, you can start receiving Interactions via webhook! At this point, your app will **no longer receive Interactions over the gateway**. If you want to receive them over the gateway again, simply delete your URL.
```

Comment:
Do you receive them _immediately_ or after you restart your bot's gateway?

Reply:
Immediately